src/lightbox/LightboxActionSheet.js

```diff
@@ -7,31 +7,31 @@ import Toast from '../utils/showToast';
 
 type DownloadImageType = {
   url: string,
-  auth: Auth,
+  auth: Auth
 };
 
 type ShareLinkType = {
-  url: string,
+  url: string
 };
 
 type ExecuteActionSheetActionType = {
   title: string,
   url: string,
-  auth: Auth,
+  auth: Auth
 };
 
 type ButtonProps = {
   auth?: Auth,
-  url: string,
+  url: string
 };
 
 type ButtonType = {
   title: string,
-  onPress: (props: ButtonProps) => void | boolean,
+  onPress: (props: ButtonProps) => void | boolean
 };
 
 const downloadImage = ({ url, auth }: DownloadImageType) => {
-  download(url, auth).then(() => Toast('Download complete.'));
+  download(url, auth, () => Toast('Download complete.'), () => Toast("Can't download"));
```

Comment:
... now you can do an `await` inside a `try { } catch {}` block and handle the error, without ever passing callbackfunctions.